src/components/SignatureRequest.js

```diff
@@ -0,0 +1,182 @@
+import React, { useEffect, useState } from 'react'
+import styled from 'styled-components'
+import { Button, GU, textStyle, useTheme } from '@aragon/ui'
+import { useWallet } from '../providers/Wallet'
+import { signMessage } from '../lib/web3-utils'
+import { dayjs } from '../utils/date-utils'
+import signRequestIllustration from '../../src/assets/signRequest.svg'
+import signRequestSuccessIllustration from '../../src/assets/signRequestSuccess.svg'
+import signRequestFailIllustration from '../../src/assets/signRequestFail.svg'
+
+const SignerRequest = React.memo(function SignerRequest({
+  actionText,
+  compactMode,
+  onActionErrorScreen,
+  onActionSuccessScreen,
+  onSignSuccessAction,
+  successText,
+  title,
+}) {
+  const [signingError, setSigningError] = useState(false)
+  const [signHash, setSignHash] = useState('')
+
+  const wallet = useWallet()
+  const theme = useTheme()
+
+  const illustration = getIllustration(signingError, signHash)
+  const { statusText, statusTextColor } = getStatusText(
+    signingError,
+    signHash,
+    theme
+  )
+  const infoText = getInfoText(signingError, signHash, successText, actionText)
+
+  const buttonWidth = compactMode ? '100% ' : `calc((100% - ${2 * GU}px) /  2)`
+
+  useEffect(() => {
+    const requestSignature = async () => {
+      if (!wallet || signingError) {
+        return
+      }
+      const now = dayjs().toString()
+      const { signHash, error } = await signMessage(wallet, now)
+
+      if (error) {
+        setSigningError(true)
+        return
+      }
+      const { error: actionError } = await onSignSuccessAction()
```

Comment:
An example of this action would be verifying the email right?

Reply:
yes, is the action that should be executed after the sign